frontend-new/src/sensitiveData/services/encryptionService/encryptionService.ts

```diff
@@ -0,0 +1,150 @@
+import { EncryptionConfig } from "src/sensitiveData/config/encryptionConfig";
+import {
+  AESResult,
+  EncryptedSensitivePersonalData,
+  EncryptionServiceOptions,
+} from "src/sensitiveData/services/encryptionService/types";
+import { getSensitivePersonalDataRSAEncryptionKey, getSensitivePersonalDataRSAEncryptionKeyId } from "src/envService";
+import { SensitivePersonalData } from "../sensitivePersonalDataService/types";
+
+// function to convert a string to an ArrayBuffer
+// from https://developers.google.com/web/updates/2012/06/How-to-convert-ArrayBuffer-to-and-from-String
+function str2ab(str: string): ArrayBuffer {
+  const buf = new ArrayBuffer(str.length);
+  const bufView = new Uint8Array(buf);
+  for (let i = 0, strLen = str.length; i < strLen; i++) {
+    bufView[i] = str.charCodeAt(i);
+  }
+  return buf;
+}
+
+/**
+ * Encryption service provides methods for encrypting sensitive data.
+ */
+export class EncryptionService {
+  private crypto: Crypto;
+
+  constructor(public options: EncryptionServiceOptions) {
+    this.crypto = options.crypto;
+    this.options = options;
+  }
+
+  private async extractRSAPublicCryptoKey(publicKeyString: string): Promise<CryptoKey> {
+    const keyData = publicKeyString
+      .replace(/-----BEGIN PUBLIC KEY-----/, "")
+      .replace(/-----END PUBLIC KEY-----/, "")
+      .replace(/\n/g, "");
+
+    const binaryDerString = window.atob(keyData);
+
+    const binaryDer = this.options.stringToBuffer(binaryDerString);
+
+    return await this.crypto.subtle.importKey(
+      EncryptionConfig.RSA.FORMAT,
+      binaryDer,
+      {
+        name: EncryptionConfig.RSA.ALGORITHM,
+        hash: EncryptionConfig.RSA.KEY_HASH_FN,
+      },
+      true, // Is extractable
+      ["encrypt"] // used for encryption here
+    );
+  }
+
+  private async encryptWithAES(plainText: string): Promise<AESResult> {
+    const encoder = new TextEncoder();
+    const dataBuffer = encoder.encode(plainText);
+
+    const cryptoKey = await this.crypto.subtle.generateKey(
```

Comment:
this can throw as well... are we handling these errors elsewhere?

Reply:
It is caught by the component. 

Because we don't want to propagatte erros.

So the first caller will be the one to handle errors because that is where we log and enque the snackbar.

That is why we agreed to do this.